### DISCUSSION TITLE
Help Me Understand the Memory Implications of `useInfiniteQuery` 🙏

### DISCUSSION THREAD

**Discussion by harrismcc** [TanStack/query#7943] [Q&A] [answered]
I'm a bit confused about the inner workings of `useInfiniteQuery` and was hoping that someone could help me understand. I'm able to implement an infinite query, even a bi-directional one, no-problem but I don't understand the memory-implications of using infinite queries.

Let's say that my api will return an infinite number of pages, and the user scrolls a TON (thus loading a lot of data). Will react-query just keep filling up until there is an OOM error? Or is it "smart" enough to handle this case and un-load unused data?

**Comment by TkDodo** [comment, marked as answer]
you need to specify `maxPages` if you want to limit the amount of pages that get cached.

https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5#limited-infinite-queries-with-new-maxpages-option
